src/en/news/blog/2022/mclock-vs-wpq-testing-with-background-ops-part2/index.md

```diff
@@ -0,0 +1,266 @@
+2. osd_scrub_cost
+
+osd_max_scrubs controls the maximum number of simultaneous scrub operations for an OSD. By default, this value is 1. For the tests in the previous sections we modified it to 5 as mentioned before.
+We will now take a look at the repercussions of making it higher.
+osd_scrub_cost is the cost assigned to the scrub operation. By default, this cost is 50 MB. With WPQ, an operation having a lower cost has a higher chance of being picked. The osd_scrub_cost works similarly with the mClock scheduler. We will be analyzing test results where the scrub cost has been decreased to 20MB.
+
+##### Increasing osd max scrubs
+
+The graphs below show the results obtained from increasing the number of concurrent scrubs to 10. As we would have expected, the total scrub duration has become half of what we saw with osd max scrubs as 5.
+
+![](images/image9_scrub_time_with_osd_max_scrubs_modified.png)
+
+The difference in the average throughput can be seen in the graph below.
+
+![](images/image10_IOPS_with_osd_max_scrubs_modified.png)
+
+A comparison between the average client latency of high client profile (osd max scrubs = 10) and WPQ shows us the increase in latency while scrub is in progress.
+
+![](images/image11_avg_clat_while_scrubbing_osd_max_scrubs_modified.png)
+
+The blue dot in the graph above indicates the point at which scrub finished for the high client profile test with max scrubs set to 10.
+
+From the above graphs we can see the effect of the number of concurrent scrubs increasing on the client throughput and latency. If we take a closer look at the numbers, the client throughput (high client profile) has decreased by 6% when compared to WPQ. The average client latency for just the duration of the scrub with the high client profile is about 7 msec. This is a 20% increase from the average client latency of WPQ (5.8 msec). Meanwhile we see a decrease of 44% in the total scrub duration.
+In conclusion, this experiment proves that increasing osd_max_scrubs to reduce scrub duration is a viable option but the user should be careful to not increase the concurrent scrubs to a very high number since this affects the client throughput and latency.
+
+##### Reducing scrub cost
+
+Every operation has a cost associated with it. As discussed in the beginning of this section, the lower the cost of the operation the higher are the chances of it getting dequeued faster. The scrub operations, by default, have a cost of 50MB.
+WPQ uses a combination of priority and cost to decide which operations are dequeued. Due to this it is easy to control operations such as scrub. For example, the osd_requested_scrub_priority is 120, this means that when a user explicitly asks for a scrub instead of waiting for the scheduled scrub WPQ is able to prioritize scrub operations over client operations (priority of 63). We are looking at the cost parameter with the hope that it could do the same for mClock.
```

Comment:
```suggestion
WPQ uses a combination of priority and cost to decide which operations are dequeued. Due to this, it is easy to control operations such as scrub. For example, the osd_requested_scrub_priority is 120. This means that when a user explicitly asks for a scrub instead of waiting for the scheduled scrub, WPQ is able to prioritize scrub operations over client operations (priority of 63). We are looking at the cost parameter with the hope that it could do the same for mClock.
```